GeoServer 2.23.0 release announcement (#143)
geoserver/geoserver.github.io: _posts/2023-04-05-geoserver-2-23-0-released.md

---
author: Jody Garnett
date: 2023-04-05
layout: post
title: GeoServer 2.23.0 Release
categories:
- Announcements
tags:
- Release
release: release_223
version: 2.23.0
jira_version: 16885
---

GeoServer [2.23.0](/release/2.23.0/) release is now available with downloads ([bin](https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.0/geoserver-2.23.0-bin.zip/download), [war](https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.0/geoserver-2.23.0-war.zip/download), [windows](https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.0/GeoServer-2.23.0-winsetup.exe/download)), along with [docs](https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.0/geoserver-2.23.0-htmldoc.zip/download) and [extensions](https://sourceforge.net/projects/geoserver/files/GeoServer/2.23.0/extensions/).

This is a stable release of GeoServer suitable for production systems, made in conjunction with GeoTools 29.0 and GeoWebCache 1.23.0.

> Thanks to Jody Garnett (GeoCat) for making this release. Additional thanks to those volunteering to test the release candidate, your assistance is seen and appreciated: Peter Rushforth, Mark Prins, Gabriel Roldan, and Juan Luis.

Author: Is this correct? I know it was a short testing window :)

@aaime you indicated that extensive testing had been done of the spring framework and spring-security update - can we thank anyone from GeoSolutions here also?

> Thanks to Jody Garnett (GeoCat) for making this release. Additional thanks to those volunteering to test the release candidate, your assistance is seen and appreciated: Peter Rushforth, Mark Prins, Gabriel Roldan, and Juan Luis.

Reviewer: ```suggestion
Thanks to Jody Garnett (GeoCat) for making this release. Additional thanks to those volunteering to test the release candidate, your assistance is seen and appreciated: Peter Rushforth, Mark Prins, Gabriel Roldan, and Juan Luis Rodríguez.
```

Reviewer 2: Probably above too.

Author: addressed, thanks! also in prior RC1 post.

Keeping GeoServer sustainable requires community commitment. If you are unable to contribute time, [sponsorship options](https://github.com/geoserver/geoserver/wiki/Sponsor) are available via the Open Source Geospatial Foundation.

### Security Considerations

> This release addresses a security vulnerabilities and is considered an essential upgrade for production systems.

Reviewer 2: ```suggestion
This release addresses a security vulnerability and is considered an essential upgrade for production systems.
```

* [GSIP-215 - Drop Java 8 Support](https://github.com/geoserver/geoserver/wiki/GSIP-215)
* [GEOS-10638](https://osgeo-org.atlassian.net/browse/GEOS-10638) Drop Java 8 support

### CSS Cleanup

The first big internal change for this release of GeoServer is a cleanup of the theme used for the GeoServer web administration application. Previously the pages had lots of little styling adjustments to try and get components to line up correctly and appear okay.

With this update all of the handmade styling changes have been removed, and everything is managed by the "geoserver.css" theme.

Thanks to Michel Gabriël (GeoCat) who started this work at the Bolsena code-sprint as a labour of love (well frustration).

* [GSIP 213 - GUI CSS Cleanup](https://github.com/geoserver/geoserver/wiki/GSIP-213)
* [GEOS-10556](https://osgeo-org.atlassian.net/browse/GEOS-10556) Cleanup Inconsistent DOM structure and use of hardcoded styles

### Spring Upgrade

> The second internal change for this release of GeoServer in an upgrade to the Spring Framework used to wire the internals of GeoServer together.

Reviewer 2: ```suggestion
The second internal change for this release of GeoServer is an upgrade to the Spring Framework used to wire the internals of GeoServer together.
```

> While this should not result in any change to functionality, it has resulted in quite a lot of carefult quality assurance and testing to ensure everything is still connected and works as intended.

Reviewer 2: ```suggestion
While this should not result in any change to functionality, it has resulted in quite a lot of careful quality assurance and testing to ensure everything is still connected and works as intended.
```

> Thanks to Joseph Miler (GeoSolution) who worked on this activity.

Reviewer 2: Maybe:

```suggestion
Thanks to Joseph Miller (GeoSolutions) who worked on this activity.
```

* [GEOS-10779](https://osgeo-org.atlassian.net/browse/GEOS-10779) Upgrade GeoServer Core Spring to 5.3.23 and Spring Security to 5.7.3

* [GEOS-10907](https://osgeo-org.atlassian.net/browse/GEOS-10907) Update spring.version from 5.3.25 to 5.3.26

### Windows installer Java 11 Update

> Windows users are advised to keep the Java 11 minium requirement in mind when upgrading existing systems.

Reviewer 2: ```suggestion
Windows users are advised to keep the Java 11 minimum requirement in mind when upgrading existing systems.
```

The installer will correctly detect the [Adoptium](https://adoptium.net) Java 17 and Java 11:

![Adoptium JDK 17 Manual](/img/posts/2.23/windows_jdk17.png) <br/>

The windows installer does not detect Oracle JDK 17; but you can use Browse to manually select this JDK:

![Oracle JDK 17 Manual](/img/posts/2.23/windows_jdk17_oracle_manual.png) <br/>

* [GEOS-10890](https://osgeo-org.atlassian.net/browse/GEOS-10890) Wrong path for the license file in the Windows installer script

> Thanks to Juan Luis (GeoCat) for troubleshooting the windows installer for this release.

Reviewer 2: And here.

### OGC CITE Fixes

The traditional OGC Open Web Services have not had automated CITE tests run for a while, but a few fixes have been made to restore CITE compliance:

* [GEOS-10787](https://osgeo-org.atlassian.net/browse/GEOS-10787) CITE WCS 1.1.1 - Throw exception on bad 'store' parameter

* [GEOS-10788](https://osgeo-org.atlassian.net/browse/GEOS-10788) CITE WCS 1.1.1 - Empty InterpolationMethod should throw exception

* [GEOS-10757](https://osgeo-org.atlassian.net/browse/GEOS-10757) CITE: WMS &lt;Style&gt; has elements in wrong order (DTD validation)

* [GEOS-10782](https://osgeo-org.atlassian.net/browse/GEOS-10782) CITE WFS 1.1 - HITS mimetype is incorrect

* [GEOS-10783](https://osgeo-org.atlassian.net/browse/GEOS-10783) CITE WFS 1.1 - Check customized feature type to determine if transform wrapper needed

* [GEOS-10784](https://osgeo-org.atlassian.net/browse/GEOS-10784) CITE WFS 1.1 - don't do illegal geometry conversions

* [GEOS-10785](https://osgeo-org.atlassian.net/browse/GEOS-10785) CITE WFS 1.1 - Data Dir - allow anonymous users to modify data

> Thanks to David Blasby (GeoCat) for this work on behalf of the GeoCat Live Project. David address several errors in the CITE testing for these services while addressing the above issues for the GeoServer community.

Reviewer 2: ```suggestion
Thanks to David Blasby (GeoCat) for this work on behalf of the GeoCat Live Project. David addressed several errors in the CITE testing for these services while addressing the above issues for the GeoServer community.
```
